flink-core/src/main/java/org/apache/flink/metrics/reporter/JMXReporter.java

```diff
@@ -73,10 +86,61 @@ public JMXReporter() {
 	// ------------------------------------------------------------------------
 
 	@Override
-	public void open(Configuration config) {}
+	public void open(Configuration config) {
+		this.jmxServer = startJmxServer(config);
+	}
+
+	private static JMXServer startJmxServer(Configuration config) {
+		JMXServer jmxServer;
+
+		String portRange = config.getString(KEY_METRICS_JMX_PORT, "9010-9025");
+		String[] ports = portRange.split("-");
+
+		if (ports.length == 0 || ports.length > 2) {
+			throw new IllegalArgumentException("JMX port range was configured incorrectly. " +
+				"Expected: <startPort>[-<endPort>] Configured: " + portRange);
+		}
+
+		if (ports.length == 1) { //single port was configured
+			int port = Integer.parseInt(ports[0]);
+			jmxServer = new JMXServer(port);
+			try {
+				jmxServer.start();
+			} catch (IOException e) {
+				throw new RuntimeException("Could not start JMX server on port " + port + ".");
+			}
+			return jmxServer;
+		} else { //port range was configured
+			int start = Integer.parseInt(ports[0]);
+			int end = Integer.parseInt(ports[1]);
+			while (true) {
+				try {
+					jmxServer = new JMXServer(start);
+					jmxServer.start();
+					LOG.info("Starting JMX on port " + start + ".");
```

Comment:
I mean it could also be interesting to see which ports where tried to start the JMX server on. Maybe as a debug logging statement.